### PR TITLE
[W5.4][T16-A2]KENNARD NG POOL HUA

### PR DESCRIPTION
**Pull request by Kennard123661**
Add edit command to AddressBook
Work in Progress until Thursday, 14th September 2017 1200hrs

### CODE REVIEW

**Comment by Kennard123661** [comment]
Work in progress because I might want to make it better. But as of now, it works fine

**Comment by Kennard123661** [comment]
ready for review

**Comment by Valerieyue** [comment]
The edit command looks good.

File: src/seedu/addressbook/parser/Parser.java [hunk 1]
```diff
@@ -138,6 +148,38 @@ private Command prepareAdd(String args) {
...
+     *
+     * @param args full command args string
+     * @return the prepared command
+     */
+    private Command prepareAdd(String args) {
+        final Matcher matcher = PERSON_DATA_ARGS_FORMAT.matcher(args.trim());
+            // Validate arg string format
+            if (!matcher.matches()) {
+        return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+    }
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/parser/Parser.java]: [Coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html) violation! One more more of these problems detected:
1. Indentation problem.

File: src/seedu/addressbook/parser/Parser.java [hunk 2]
```diff
@@ -138,6 +148,38 @@ private Command prepareAdd(String args) {
...
+                        matcher.group("phone"),
+                        isPrivatePrefixPresent(matcher.group("isPhonePrivate")),
+
+                        matcher.group("email"),
+                        isPrivatePrefixPresent(matcher.group("isEmailPrivate")),
+
+                        matcher.group("address"),
+                        isPrivatePrefixPresent(matcher.group("isAddressPrivate")),
+
+                        getTagsFromArgs(matcher.group("tagArguments"))
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/parser/Parser.java]: Perhaps refactoring of magic numbers would reduce potential errors when more field are added.

**Pull request closed without merging**